fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1155,7 +1155,13 @@ private PredicateSplit getCompensationPredicates(RewriteContext rewriteContext,
         if (srcPu == null && targetPu != null) {
             // query: empid < 5
             // mv: empid < 5 or salary > 100
-            srcPu = Utils.compoundAnd(compensationEqualPredicate, compensationPr);
+            if (!isQueryAgainstView) {
+                // compensationEqualPredicate and compensationPr is based on query, need to change it to view based
```

Comment:
Why ViewAgainsQuery need to rewrite to view based?

Reply:
because compensationPr is now based on query, but we need predicates based on views as input